### PR TITLE
fix(jit): input arguments required error

### PR DESCRIPTION
**Pull request by karatakis**
**Summary:**  
This PR will enable Tailcall JIT implementation to validate the Input arguments if their fields meet the schema requirements. When an argument does not meet the requirements, an error is thrown.

**Fixes:**
* Close: https://github.com/tailcallhq/tailcall/issues/2809

**Build & Testing:**

- [x] I ran `cargo test` successfully.
- [x] I have run `./lint.sh --mode=fix` to fix all linting issues raised by `./lint.sh --mode=check`.

**Checklist:**

- [x] I have added relevant unit & integration tests.
- [X] I have updated the [documentation] accordingly.
- [x] I have performed a self-review of my code.
- [X] PR follows the naming convention of `<type>(<optional scope>): <title>`

[documentation]: https://github.com/tailcallhq/tailcallhq.github.io/tree/develop/docs

### CODE REVIEW

File: src/core/jit/input_resolver.rs [hunk 1]
```diff
@@ -66,4 +100,79 @@ where
...
+            if !type_of.is_nullable() && value.is_none() {
+                let default_value = default_value.clone();
+                match default_value {
+                    Some(value) => Ok(Some(value)),
+                    None => Err(ResolveInputError::ArgumentIsRequired {
+                        arg_name: arg_name.to_string(),
+                        field_name: parent_name.to_string(),
+                    }),
+                }
+            } else if !type_of.is_nullable() && is_value_null {
```
Comment (meskill) [inline src/core/jit/input_resolver.rs:124]: pile of matches is hard to read sometimes. Wdyt to use `let-else` and early returns instead, will it be more readable?

File: src/core/jit/input_resolver.rs [hunk 2]
```diff
@@ -66,4 +100,79 @@ where
...
+                    } else if let Some(arr) = value.as_array_mut() {
+                        for (index, item) in arr.iter_mut().enumerate() {
+                            let parent_name = format!("{}.{}.{}", parent_name, arg_name, index);
+
+                            *item = self
+                                .recursive_parse_arg(
+                                    &parent_name,
+                                    &index.to_string(),
+                                    type_of,
+                                    default_value,
```
Comment (meskill) [inline src/core/jit/input_resolver.rs:165]: is default value is even applicable for the arrays? we can only define defaults only for argument or field in input type. Should be `None`?

Comment (karatakis) [reply src/core/jit/input_resolver.rs:165]: Good catch, its always None

File: src/core/jit/input_resolver.rs [hunk 3]
```diff
@@ -66,4 +88,86 @@ where
...
+            value
+        } else {
+            return Ok(None);
+        };
+
+        let def = if let Some(def) = self.plan.index.get_input_type_definition(type_of.name()) {
+            def
+        } else {
+            return Ok(Some(value));
+        };
```
Comment (meskill) [inline src/core/jit/input_resolver.rs:134]:
```suggestion
        let Some(mut value) = value? else {
            return Ok(None);
        };

        let Some(def) = self.plan.index.get_input_type_definition(type_of.name()) else {
            return Ok(Some(value));
        };
```

File: src/core/jit/input_resolver.rs [hunk 4]
```diff
@@ -66,4 +88,86 @@ where
...
+                Err(ResolveInputError::ArgumentIsRequired {
+                    arg_name: arg_name.to_string(),
+                    field_name: parent_name.to_string(),
+                })
+            } else if value.is_none() {
+                let default_value = default_value.clone();
+                Ok(default_value)
+            } else {
+                Ok(value)
+            };
```
Comment (meskill) [inline src/core/jit/input_resolver.rs:122]:
```suggestion
        let value = if !type_of.is_nullable() && value.is_none() {
            let default_value = default_value.clone();

            Some(default_value.ok_or(ResolveInputError::ArgumentIsRequired {
                arg_name: arg_name.to_string(),
                field_name: parent_name.to_string(),
            })?)
        } else if !type_of.is_nullable() && is_value_null {
            return Err(ResolveInputError::ArgumentIsRequired {
                arg_name: arg_name.to_string(),
                field_name: parent_name.to_string(),
            });
        } else if value.is_none() {
            let default_value = default_value.clone();
            default_value
        } else {
            value
        };
```

**Review by meskill — APPROVED** [review]

**Review by meskill — APPROVED** [review]

**Review by meskill — APPROVED** [review]

**Pull request merged**